### PR TITLE
Use TypeScript 2.0

### PR DESCRIPTION
**Pull request by TsvetanMilanov**
Update the TS related dependencies to use TS 2.0 and fix the errors from
the update

### CODE REVIEW

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by rosen-vladimirov** [comment]
run ci

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by rosen-vladimirov** [comment]
run ci

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by rosen-vladimirov** [comment]
run ci

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by rosen-vladimirov** [comment]
run ci

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:heart:

File: test/unit-tests/analytics-service.ts [hunk 1]
```diff
@@ -511,37 +513,41 @@ describe("analytics-service", () => {
...
 			service = testInjector.resolve("analyticsService");
+			osInfo = testInjector.resolve("osInfo");
+			osType = osInfo.type;
+			osRelease = osInfo.release;
 		});
 
 		after(() => {
-			os.type = osType;
-			os.release = osRelease;
+			osInfo.type = osType;
```
Comment (rosen-vladimirov) [inline test/unit-tests/analytics-service.ts]: the `after` task should return everything in the correct way, so other tests will use the correct osInfo. In this case we set the `type` property of osInfo to something that is not the default value. Is this correct?

File: bootstrap.ts [hunk 2]
```diff
@@ -10,6 +10,7 @@ $injector.require("errors", "./errors");
 $injector.requirePublic("fs", "./file-system");
 $injector.require("sysInfoBase", "./sys-info-base");
 $injector.require("hostInfo", "./host-info");
+$injector.require("osInfo", "./os-info");
```
Comment (rosen-vladimirov) [inline bootstrap.ts]: Ok, the interface will be IoS, which is totally strange and will be confused with iOS, so osInfo is ok

File: prompter.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import Future = require("fibers/future");
-import * as prompt from "inquirer";
+import prompt = require("inquirer");
```
Comment (rosen-vladimirov) [inline prompter.ts]: sure?

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:heart:

**Pull request merged**